testing/src/main/java/org/oppia/android/testing/espresso/TextInputAction.kt

```diff
@@ -0,0 +1,45 @@
+package org.oppia.android.testing.espresso
+
+import android.view.View
+import androidx.test.espresso.matcher.BoundedMatcher
+import com.google.android.material.textfield.TextInputLayout
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+
+/**
+ * Action for matching error text and performing other actions on TextInputLayout
+ * in a test infrastructure-specific way.
+ */
+class TextInputAction {
```

Comment:
I tried a bit more cleaner way to do this. Currently, the function which we call directly `hasErrorText` and `hasNoErrorText` is the one that contains the whole logic of testing. I tried to abstract it out. 

```kt
  companion object {
    /**
     * Check that the TextInputLayout view has the [expectedErrorText] error text.
     *
     * @returns a [BoundedMatcher] that matches the [expectedErrorText] with the
     * TextInputLayout's error text
     */
    fun hasErrorText(expectedErrorText: String): BoundedMatcher<View, TextInputLayout> {
      return ErrorTextExisted(expectedErrorText)
    }

    /**
     * Returns a [Matcher] that matches if the textInputLayout has no error text available.
     */
    fun hasNoErrorText(): BoundedMatcher<View, TextInputLayout> {
      return ErrorTextNotExisted()
    }

    private class ErrorTextExisted(private val expectedErrorText: String) :
      BoundedMatcher<View, TextInputLayout>(TextInputLayout::class.java) {
      override fun describeTo(description: Description) {
        description.appendText("The expected error text is '${expectedErrorText}'")
      }

      override fun matchesSafely(textInputLayout: TextInputLayout): Boolean {
        return (textInputLayout).error == expectedErrorText
      }
    }

    private class ErrorTextNotExisted :
      BoundedMatcher<View, TextInputLayout>(TextInputLayout::class.java) {
      override fun matchesSafely(textInputLayout: TextInputLayout): Boolean {
        return (textInputLayout).error.isNullOrEmpty()
      }

      override fun describeTo(description: Description) {
        description.appendText("")
      }
    }
  }
```